### PR TITLE
fix hessian2 反序列化bug #3640

### PR DESCRIPTION
**Pull request by CrazyHZM**
修复https://github.com/apache/incubator-dubbo/issues/3640 问题

### CODE REVIEW

**Comment by lovepoem** [comment]
Please see the java7 error https://travis-ci.org/dubbo/hessian-lite/jobs/506137514

**Comment by CrazyHZM** [comment]
> Please see the java7 error https://travis-ci.org/dubbo/hessian-lite/jobs/506137514


I will look later, because I don't have a java7 environment yet, would you give me some advice?

**Comment by lovepoem** [comment]
>  I will look later, because I don't have a java7 environment yet, would you give me some advice?

I think you can setup a java7 in your pc 

Additional,  please add some test case to verify your change for the issue: https://github.com/apache/incubator-dubbo/issues/3640

**Comment by CrazyHZM** [comment]
> > I will look later, because I don't have a java7 environment yet, would you give me some advice?
> 
> I think you can setup a java7 in your pc
> 
> Additional, please add some test case to verify your change for the issue: [apache/incubator-dubbo#3640](https://github.com/apache/incubator-dubbo/issues/3640)

Yes,I will add test case later.

**Comment by CrazyHZM** [comment]
> > I will look later, because I don't have a java7 environment yet, would you give me some advice?
> 
> I think you can setup a java7 in your pc
> 
> Additional, please add some test case to verify your change for the issue: [apache/incubator-dubbo#3640](https://github.com/apache/incubator-dubbo/issues/3640)


I tested it in java7 in my local environment and found that it was passed. I don’t have any ideas for now. Let’s take another look.

**Comment by lovepoem** [comment]
LGTM

File: src/main/java/com/alibaba/com/caucho/hessian/io/Hessian2Input.java [hunk 1]
```diff
@@ -90,6 +90,7 @@
     private static final int GAP = 16;
     private static Field _detailMessageField;
     private static boolean _isCloseStreamOnClose;
+    public static boolean _isNull = false;
```
Comment (zxiang179) [inline src/main/java/com/alibaba/com/caucho/hessian/io/Hessian2Input.java]: 使用静态全局变量，在多线程场景中是否会存在线程安全问题？

**Comment by fingthinking** [comment]
这逻辑根本不对，多线程的时候，直接报错。

**Pull request merged**